### ISSUE TITLE
IFT Method selection

### ISSUE THREAD

**Issue by vlevantovsky** [w3c/IFT#56]
Considering the data we collected during PFE evaluation work on how network conditions affect the font transfer efficiency, and assuming that servers supporting both patch-subset and range-request methods would eventually learn how to optimize data transfer based on those conditions - should we allow a server to suggest a recommended approach?

In the current RangeRequest draft (#47), IFT method negotiation allows a client to request `patch-subset` method while a server may respond with `Accept-Ranges` header thereby making a recommendation to use an alternative approach to what a client has asked for. I believe this is a very useful functionality. However, there is no provision for a server to recommend another approach when a client initially asks for the `range-request` method but the server considers the `patch-subset` mechanism to be a better choice - we need to allow this to happen.

**Comment by litherum** [comment]
It seems reasonable for servers to be able to express their preference. Clients should not be required to honor that preference, though.

I think my biggest question is how, mechanically, a server could express its preference without requiring an additional synchronous round-trip before any loading can begin. The patch-subset method requires the initial request to include a particular data structure, so how can the server give information to the client before the client makes its first request?

If, instead, the initial _response_ was augmented (maybe with an additional HTTP header or something, send _in addition_ to whichever initial response the server was going to send anyway):
1. You're already partway done with whichever method you're using. Are you really going to want to start over from scratch?
2. There's already way for a client to say "I'm requesting both methods at once; respond with the response for whichever method you prefer." So, in the situation where the client was ambivalent, there's no problem - the server can already express (and enforce!) its preference. Therefore, this proposal would only be useful when the client prefers one method but the server prefers another. If the client prefers one method but the server prefers another, why would anyone expect the client to change its mind at this point? The client knows more about the network conditions than the server.

**Comment by garretrieger** [comment]
I agree with Myles. Changing the method mid transfer would be problematic, and the existing first request specification allows the server to choose one method or the other.

**Comment by vlevantovsky** [comment]
I agree. I realize that server's ability to respond to the incremental transfer request with a chunk of font data that is immediately usable (potentially enabling a first page render to occur before any subsequent roundtrip request is made) is valuable from user experience POV, and that changing the transfer method (and triggering the need for an additional request) would likely negate any hypothetical advantages.

It seems that the only viable option for a server (that _apriori_ knows more about the font that a UA does) is to either honor the original IFT method request or to reply with the whole font download (if the server believes this is the best way forward).

**Comment by vlevantovsky** [comment]
This discussion will continue as part of the issue #63 , closing.
